src/cript/api/api.py

```diff
@@ -402,7 +424,114 @@ def api_prefix(self):
     def api_version(self):
         return self._api_version
 
-    def save(self, project: Project) -> None:
+    # def save(self, project: Project) -> None: #changed bc we need it to be node agnostic
+    def save(self, new_node: PrimaryBaseNode) -> None:
+        # Member function of API
+        ################################################################################
+
+        # In this mockup, we can't actually do the search:
+        # print("old_node_paginator = self.search(node_type=cript.Project, search_mode=cript.SearchModes.UUID, value_to_search=project)")
+        # print("old_node_paginator.auto_load_nodes = False")
+
+        # node_type=Project this needs to be dynamically loaded with locals ?
+
+        # NOTE: Needed to make it node agnostic
+        node_class_name = new_node.node_type.capitalize()
+        NodeClass = getattr(PrimaryNodes, node_class_name)
+
+        # print(node_class_name)
+        # print(NodeClass)
+        # print("----------+++++-----------")
+
+        old_node_paginator = self.search(node_type=NodeClass, search_mode=SearchModes.UUID, value_to_search=str(new_node.uuid))
+        old_node_paginator.auto_load_nodes = False
+        try:
+            # print("old_node_json = next(old_node_paginator)")
+
+            # # Just for this mock-up we load from the file instead
+            # with open("new_project.json") as json_handle:
+            #     old_node_json = json_handle.read()
+            old_node_json = next(old_node_paginator)
+
+        except StopIteration:  # New Project do POST instead
+            # Do the POST request call. only on project
+            # or else its a patch handled by previous node
+
+            print("--------900---------")
+            print(new_node.node_type.lower())
+            if new_node.node_type.lower() == "project":
+                data = new_node.get_json().json
+                response = self._capsule_request(url_path="/project/", method="POST", data=data)
+                print("----700----")
+                print(response.json())
```

Comment:
@InnocentBug 

Hey Ludwig 
this is giving me the "bad uuid" for a project when I run "test inventory"

I get this out put 


```
--------900---------
project
----700----
{'code': 400, 'data': None, 'error': 'Bad uuid: 9aa4b9e7-1b7c-4c76-9602-2c980bc4298a provided'}
```

<img width="475" alt="Screenshot 2024-04-02 at 5 43 51 PM" src="https://github.com/C-Accel-CRIPT/Python-SDK/assets/20729581/85a01130-f968-4533-bfe4-654c9f862c65">

Reply:
That is probably the same issue that we discussed before.

The API requires all materials to be known before an inventory can be saved.
But in our tests, we have inventories with new materials in them.

So, you get a bad UUID, when you try to save one.

We will have to redesign the fixtures, such that they contain known materials.

Reply:
for now, we can ignore it, I guess.

Or better, change the inventories, such that they contain no or known materials.

But we also have to think about the user, parse this error, and give a good error message, when this happens.